File: 30-Day LeetCoding Challenge/Python/lastStoneWeight.py
class Solution(object):
    def lastStoneWeight(self, stones):
        """
        :type stones: List[int]
        :rtype: int
        """
        if len(stones) == 0:
            return 0
        elif len(stones) == 1:
            return stones[0]
        else:
            sorted_stones = sorted(stones)
            new_stone = sorted_stones[-1] - sorted_stones[-2]
            new_stones = sorted_stones[:-2] + [new_stone]
            return self.lastStoneWeight(new_stones)

input = [2,7,4,1,8,1]
s = Solution()
print(s.lastStoneWeight(input))

File: 30-Day LeetCoding Challenge/C++/lastStoneWeight.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        if(stones.size() == 0) return 0;
        else if(stones.size() == 1) return stones[0];
        else {
            sort(stones.begin(), stones.end());
            int new_stone = stones[stones.size()-1] - stones[stones.size()-2];
            stones.pop_back();
            stones.pop_back();
            stones.push_back(new_stone);
            return lastStoneWeight(stones);
        } 
    }
};
int main() {
    Solution s;
    vector<int> input = {2,7,4,1,8,1};
    cout << s.lastStoneWeight(input);
    return 0;
}